lib/src/utils.dart

```diff
@@ -12,11 +12,11 @@ import 'byte_stream.dart';
 ///
 ///     mapToQuery({"foo": "bar", "baz": "bang"});
 ///     //=> "foo=bar&baz=bang"
-String mapToQuery(Map<String, String> map, {Encoding encoding}) {
+String mapToQuery(Map<String, String> map, {Encoding? encoding}) {
   var pairs = <List<String>>[];
   map.forEach((key, value) => pairs.add([
-        Uri.encodeQueryComponent(key, encoding: encoding),
-        Uri.encodeQueryComponent(value, encoding: encoding)
+        Uri.encodeQueryComponent(key, encoding: encoding ?? utf8),
```

Comment:
`Uri.encodeQueryComponent` `encoding` param is now non-null with a default.

So we have to hoist the default value here for a `null` `encoding`